runtime/engine/src/main/java/io/aklivity/zilla/runtime/engine/internal/registry/FileWatcherTask.java

```diff
@@ -27,44 +28,45 @@
 import java.nio.file.FileSystems;
 import java.nio.file.WatchKey;
 import java.nio.file.WatchService;
-import java.security.MessageDigest;
-import java.security.NoSuchAlgorithmException;
 import java.util.IdentityHashMap;
 import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
 import java.util.function.BiFunction;
 
 import io.aklivity.zilla.runtime.engine.config.NamespaceConfig;
 
 public class FileWatcherTask extends WatcherTask
 {
     private final Map<WatchKey, WatchedConfig> watchedConfigs;
-    private final MessageDigest md5;
     private final WatchService watchService;
 
     public FileWatcherTask(
         BiFunction<URL, String, NamespaceConfig> changeListener)
     {
         super(changeListener);
         this.watchedConfigs = new IdentityHashMap<>();
-        MessageDigest md5 = null;
         WatchService watchService = null;
 
         try
         {
-            md5 = MessageDigest.getInstance("MD5");
             watchService = FileSystems.getDefault().newWatchService();
         }
-        catch (NoSuchAlgorithmException | IOException ex)
+        catch (IOException ex)
         {
             rethrowUnchecked(ex);
         }
 
-        this.md5 = md5;
         this.watchService = watchService;
 
     }
 
+    @Override
+    public Future<Void> submit()
+    {
+        return commonPool().submit(this);
```

Comment:
Is this already fixed in the other PR, if not then we need to address it here to avoid using `commonPool()`.